buildSrc/src/main/kotlin/io/spine/internal/gradle/ConfigTester.kt

```diff
@@ -302,31 +311,31 @@ class ClonedRepo(
     @Suppress("TooGenericExceptionCaught")
     private fun copyFolder(sourceFolder: Path, ignoredFolder: Path?, destinationFolder: Path) {
         try {
-            Files.walk(sourceFolder).forEach { file: Path ->
-                if (ignoredFolder != null) {
-                    if (file.toAbsolutePath().startsWith(ignoredFolder.toAbsolutePath())) {
-                        return@forEach
-                    }
-                }
-                try {
-                    val destination = destinationFolder.resolve(sourceFolder.relativize(file))
-                    if (Files.isDirectory(file)) {
-                        if (!Files.exists(destination)) Files.createDirectory(destination)
-                        return@forEach
-                    }
-                    Files.copy(file, destination)
-                } catch (e: Exception) {
-                    throw IllegalStateException(
-                        "Error copying folder `$sourceFolder` to `$destinationFolder`.", e
-                    )
-                }
-            }
+            copyRecursively(sourceFolder, ignoredFolder, destinationFolder)
         } catch (e: Exception) {
             throw IllegalStateException(
                 "Error copying folder `$sourceFolder` to `$destinationFolder`.", e
             )
         }
     }
+
+    private fun copyRecursively(sourceFolder: Path, ignoredFolder: Path?, destinationFolder: Path) {
```

Comment:
Don't we have have it natively available from Kotlin?

Reply:
Yes, partially. Kotlin has `File.copyRecursively()` but it doesn't allow specifying exclusions. We could copy the whole directory and then delete asked `ignoredFolder` but it would come with some penalty in case `ignoredFolder` is big.